### PR TITLE
[WIP] Manifestos

### PR DESCRIPTION
**Pull request by Joannis**
The goal of this PR is to write a manifesto for every officially supported module.

Each manifesto will cover the goals and scope of the project, in addition to details of how this is achieved and why.

In addition to the module manifestos, a manifesto is written for minimum code quality definition. No tag may be placed without fully conforming to the relevant manifesto(s).

### CODE REVIEW

File: Documents/Manifestos/ASYNC.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+- Event
+- Error
+- End of events
+
+The end of events marks the end of a stream. Like a TCP socket being closed, or (in the real world) a faucet being closed.
+
+## Why
+
+In Swift there is currently no library that provides an easy, accessible and comprehensible way to work with asynchronous data flows.
```
Comment (vzsg) [inline Documents/Manifestos/ASYNC.md]: RX (including, but not limited to, RxSwift) users might beg to differ on this statement.

Comment (Joannis) [reply Documents/Manifestos/ASYNC.md]: I see, they support Swift on Linux nowadays.

File: Documents/Manifestos/ASYNC.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+This library needs to be simple and independent, so that it can be easily and without consideration be used in foreign libraries (outside of the Vapor project).
+
+## Scope
+
+Anything not directly a goal is out of scope.
+
+## API Goals
+
+To provide a clean, readable and comprehensible way of writing asynchronous code.
```
Comment (tanner0101) [inline Documents/Manifestos/ASYNC.md]: comma after readable

File: Documents/Manifestos/ASYNC.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+- Event
+- Error
+- End of events
+
+The end of events marks the end of a stream. Like a TCP socket being closed, or (in the real world) a faucet being closed.
+
+## Why
+
+Asynchronous is important for a webframework like vapor and it's components to keep all components both scalable and maintainable.
```
Comment (tanner0101) [inline Documents/Manifestos/ASYNC.md]: `webframework` -> `web framework`. `it's` - > `its`

File: Documents/Manifestos/CODE_QUALITY.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
+# Code Quality manifesto
+
+Code quality needs to be ensured for every project. Projects the size of Vapor, however, need strict checks to ensure there is little to no quality regression.
```
Comment (tanner0101) [inline Documents/Manifestos/CODE_QUALITY.md]: no commas needed around `however`

File: Documents/Manifestos/CODE_QUALITY.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+This manifesto is designed as a base line for code quality in all official Vapor projects.
```
Comment (tanner0101) [inline Documents/Manifestos/CODE_QUALITY.md]: baseline one word

File: Documents/Manifestos/CODE_QUALITY.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+## What defines quality code?
+
+Code quality is a very broad term that can mean many things. Definitions differ per person. For this reason, we define the following as quality code:
+
+**Unit tests:**
+
+- An absolute minimum of 80% test coverage
+- When critical code is dependent on random factors, these random factors will be simulated to the extend possible
```
Comment (tanner0101) [inline Documents/Manifestos/CODE_QUALITY.md]: extend -> extent

**Pull request merged**